lib/src/defaults/providers/firebase/firebase_external_interface.dart

```diff
@@ -28,15 +26,21 @@ class FirebaseExternalInterface
     on<FirebaseWatchAllRequest>(_withFirebaseWatchAllRequest);
   }
 
+  @override
+  FailureResponse onError(Object error) {
```

Comment:
Downcasting to just objects instead of using Either is a downgrade to using what Functional programming is supposed to fix, I might need to hear more about why you like this more.